packages/core/src/redux/tests/store/signalConnection.spec.ts

```diff
@@ -18,70 +13,67 @@ jest.mock("@whereby.com/media", () => {
     };
 });
 
-describe("actions", () => {
-    it("doSignalSocketConnect", () => {
-        const store = createStore();
+describe("signalConnectionSlice", () => {
+    describe("actions", () => {
+        it("doSignalSocketConnect", () => {
+            const store = createStore();
 
-        store.dispatch(doSignalSocketConnect());
+            store.dispatch(doSignalSocketConnect());
 
-        expect(ServerSocket).toHaveBeenCalledTimes(1);
-        expect(mockServerSocket.connect).toHaveBeenCalledTimes(1);
-    });
-
-    it("doSignalIdentifyDevice", () => {
-        const deviceCredentials = randomDeviceCredentials();
-        const store = createStore({ withSignalConnection: true });
-
-        const before = store.getState().signalConnection;
-
-        store.dispatch(doSignalIdentifyDevice({ deviceCredentials }));
-
-        const after = store.getState().signalConnection;
-
-        expect(mockSignalEmit).toHaveBeenCalledWith("identify_device", { deviceCredentials });
-        expect(diff(before, after)).toEqual({
-            isIdentifyingDevice: true,
+            expect(ServerSocket).toHaveBeenCalledTimes(1);
+            expect(mockServerSocket.connect).toHaveBeenCalledTimes(1);
         });
-    });
 
-    it("doSignalDisconnect", () => {
-        const store = createStore({ withSignalConnection: true });
+        it("doSignalIdentifyDevice", () => {
+            const deviceCredentials = randomDeviceCredentials();
+            const store = createStore({
+                initialState: {
+                    signalConnection: {
+                        isIdentifyingDevice: false,
+                        deviceIdentified: false,
+                        status: "connected",
+                        // eslint-disable-next-line @typescript-eslint/no-explicit-any
+                        socket: mockServerSocket as any,
```

Comment:
I don't love this. But I see why you needed to it. I'll create a ticket for cleaning up this, you should be able to pass a partial initial state for signal + the `withSignalConnection` parameter.